crates/core/src/client/mod.rs

```diff
@@ -470,6 +466,27 @@ impl KakarotProvider for KakarotClient<JsonRpcClient<HttpTransport>> {
                     let eth_tx = starknet_tx.to_eth_transaction(self, None, None, None).await?;
                     let from = eth_tx.from;
                     let to = eth_tx.to;
+                    let contract_address = match to {
+                        Some(address) => Some(address),
```

Comment:
If to is Some(), then it's not a contract deployment, then it should be None

Reply:
`/// Contract address created, or None if not a deployment.
    pub contract_address: Option<Address>,`